From 455796818d521a905c3fcb7621e0cc62a94bcd04 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Mon, 19 Aug 2019 15:30:36 +0800
Subject: [PATCH] drm: fix some no need warning

As rockchip maybe appear one crtc connect two connecter,
the conn_state->best_encoder always isn't NULL, so
here no need warning.

Change-Id: I3e4274e64f4c2b4eea5d9700cebffb716425426b
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/drm_atomic_helper.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index ab99fecc69dd..a5ce13c9368f 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -217,8 +217,12 @@ set_best_encoder(struct drm_atomic_state *state,
 		 * As an exception restoring duplicated atomic state
 		 * during resume is allowed, so don't warn when
 		 * best_encoder is equal to encoder we intend to set.
+		 *
+		 * As rockchip maybe appear one crtc connect two connecter,
+		 * the conn_state->best_encoder always isn't NULL, so here no
+		 * need warning.
 		 */
-		WARN_ON(!crtc && encoder != conn_state->best_encoder);
+		WARN_ON(!crtc && encoder && encoder != conn_state->best_encoder);
 		if (crtc) {
 			crtc_state = drm_atomic_get_new_crtc_state(state, crtc);
 
-- 
2.35.3

